Iceberg REST catalog with SIGV4 fails: NoClassDefFoundError org.apache.iceberg.aws.AwsProperties (missing AWS SDK kms)

## Description

Any Iceberg REST catalog configured with `iceberg.rest-catalog.security=SIGV4` fails on first use with:

```
java.lang.NoClassDefFoundError: Could not initialize class org.apache.iceberg.aws.AwsProperties
```

The catalog loads at startup, but the error is thrown the first time the catalog is initialized by a query (e.g. `SHOW SCHEMAS`).

## Root cause

The SigV4 REST signing path loads `org.apache.iceberg.aws.AwsProperties`. As of Iceberg 1.11.0 (bundled in Trino 482), that class has a static field initializer that references a KMS SDK class:

```java
// org.apache.iceberg.aws.AwsProperties (Iceberg 1.11.0), line ~223
public static final EncryptionAlgorithmSpec KMS_ENCRYPTION_ALGORITHM_SPEC_DEFAULT =
    EncryptionAlgorithmSpec.SYMMETRIC_DEFAULT;
```

`software.amazon.awssdk.services.kms.model.EncryptionAlgorithmSpec` comes from the AWS SDK `kms` artifact, which `plugin/trino-iceberg/pom.xml` declares with `<scope>test</scope>`. It is therefore not packaged in the shipped plugin. When `AwsProperties` is loaded, its `<clinit>` throws `NoClassDefFoundError`, and the JVM then reports "Could not initialize class org.apache.iceberg.aws.AwsProperties" on every subsequent reference.

Full trace:

```
Caused by: java.lang.ClassNotFoundException: software.amazon.awssdk.services.kms.model.EncryptionAlgorithmSpec
	at org.apache.iceberg.aws.AwsProperties.<clinit>(AwsProperties.java:223)
	at org.apache.iceberg.aws.RESTSigV4AuthManager.initSession(RESTSigV4AuthManager.java:53)
	at org.apache.iceberg.rest.RESTSessionCatalog.initialize(RESTSessionCatalog.java:210)
	at io.trino.plugin.iceberg.catalog.rest.TrinoIcebergRestCatalogFactory.create(TrinoIcebergRestCatalogFactory.java:123)
	...
```

## Minimal reproduction

Stock `trinodb/trino:482`, no real AWS access required. Catalog file:

```properties
connector.name=iceberg
iceberg.catalog.type=rest
iceberg.rest-catalog.uri=https://example.invalid/iceberg
iceberg.rest-catalog.security=SIGV4
iceberg.rest-catalog.signing-name=glue
fs.s3.enabled=true
s3.region=eu-west-1
s3.aws-access-key=dummy
s3.aws-secret-key=dummy
```

Then run `SHOW SCHEMAS FROM iceberg;` → `NoClassDefFoundError`.

Confirmed: mounting `kms-2.46.17.jar` (the same AWS SDK version as the rest of the plugin) into the iceberg plugin directory resolves it — the error advances from class-init to a plain `UnknownHostException` on the bogus URI, proving the missing jar is the sole blocker.

## Affected versions

482 (and current `master` — the `kms` dependency is still `test`-scoped).

## Fix

Change the AWS SDK `kms` dependency in `plugin/trino-iceberg/pom.xml` from `test` to `runtime` scope so it is packaged with the plugin. PR incoming.

--- a/plugin/trino-iceberg/pom.xml
+++ b/plugin/trino-iceberg/pom.xml
@@ -401,6 +401,12 @@
         </dependency>
 
         <dependency>
+            <groupId>software.amazon.awssdk</groupId>
+            <artifactId>kms</artifactId>
+            <scope>runtime</scope>
+        </dependency>
+
+        <dependency>
             <groupId>com.azure</groupId>
             <artifactId>azure-security-keyvault-keys</artifactId>
             <scope>test</scope>
@@ -725,12 +731,6 @@
         <dependency>
             <groupId>org.testcontainers</groupId>
             <artifactId>testcontainers-postgresql</artifactId>
-            <scope>test</scope>
-        </dependency>
-
-        <dependency>
-            <groupId>software.amazon.awssdk</groupId>
-            <artifactId>kms</artifactId>
             <scope>test</scope>
         </dependency>
 

